### PR TITLE
Allow users manage own access keys

### PR DESCRIPTION
**Pull request by goruha**
## What
* Allow users to manage own access keys

## Why
* If they can manage passwords, they should be able to manage access keys

### CODE REVIEW

**Comment by osterman** [comment]
Fix CI

File: main.tf [hunk 1]
```diff
@@ -114,6 +137,12 @@ resource "aws_iam_policy" "allow_change_password_admin" {
   policy      = "${data.aws_iam_policy_document.allow_change_password.json}"
 }
 
+resource "aws_iam_policy" "allow_manage_access_keys_admin" {
+  name        = "${module.admin_label.id}-permit-manage-keys"
+  description = "Allow admin users to manage own access keys"
```
Comment (osterman) [inline main.tf]:
```suggestion
  description = "Allow admin users to manage their own access keys"
```

File: main.tf [hunk 2]
```diff
@@ -100,6 +103,26 @@ data "aws_iam_policy_document" "allow_change_password" {
   }
 }
 
+data "aws_iam_policy_document" "allow_manage_access_keys" {
```
Comment (osterman) [inline main.tf]:
```suggestion
data "aws_iam_policy_document" "allow_key_management" {
```

File: main.tf [hunk 3]
```diff
@@ -114,6 +137,12 @@ resource "aws_iam_policy" "allow_change_password_admin" {
   policy      = "${data.aws_iam_policy_document.allow_change_password.json}"
 }
 
+resource "aws_iam_policy" "allow_manage_access_keys_admin" {
```
Comment (osterman) [inline main.tf]:
```suggestion
resource "aws_iam_policy" "allow_key_management" {
```

File: main.tf [hunk 4]
```diff
@@ -114,6 +137,12 @@ resource "aws_iam_policy" "allow_change_password_admin" {
...
+  name        = "${module.admin_label.id}-permit-manage-keys"
```
Comment (osterman) [inline main.tf]:
```suggestion
  name        = "${module.admin_label.id}-allow-key-management"
```

Suggest using new `label` to follow convention and enforce use of `var.delimiter`

File: main.tf [hunk 5]
```diff
@@ -177,6 +211,12 @@ resource "aws_iam_policy" "allow_change_password_readonly" {
   policy      = "${data.aws_iam_policy_document.allow_change_password.json}"
 }
 
+resource "aws_iam_policy" "allow_manage_access_keys_readonly" {
+  name        = "${module.readonly_label.id}-permit-manage-keys"
+  description = "Allow readonly users to manage own access keys"
```
Comment (osterman) [inline main.tf]:
```suggestion
  description = "Allow readonly users to manage their own access keys"
```

File: main.tf [hunk 6]
```diff
@@ -151,6 +180,11 @@ resource "aws_iam_group_policy_attachment" "allow_chage_password_admin" {
   policy_arn = "${aws_iam_policy.allow_change_password_admin.arn}"
 }
 
+resource "aws_iam_group_policy_attachment" "manage_access_key_admin" {
```
Comment (osterman) [inline main.tf]:
```suggestion
resource "aws_iam_group_policy_attachment" "key_management" {
```

**Comment by osterman** [comment]
Fix CI

**Review by osterman — CHANGES_REQUESTED** [review]
They can already do this, pretty sure. They just need to assume role first. They *should not* be able to create any access keys without MFA. The changes in this PR don't seem to require MFA.  The password stuff was done (without MFA) so they can self-register after receiving an invitation link. The same is not true for keys.

**Comment by goruha** [comment]
> They can already do this, pretty sure.

Nope. Only admin roles can manage the keys.

> They just need to assume role first.

Assume role - readonly allow only to read, not to renew or create another one.

>  They should not be able to create any access keys without MFA.

Done.

> The changes in this PR don't seem to require MFA. 

Done

> The password stuff was done (without MFA) so they can self-register after receiving an invitation link. > > The same is not true for keys.

Done

**Review by osterman — APPROVED** [review]

**Review by aknysh — APPROVED** [review]

**Pull request merged**